src/powershell/Public/ConvertTo-FinOpsSchema/ConvertTo-FinOpsSchema.ps1

```diff
@@ -0,0 +1,221 @@
+    $columnsToAdd = @(
+        'ftk_AccountName', 
+        'ftk_AccountId', 
+        'ftk_AccountOwnerId', 
+        'ftk_AvailabilityZone', 
+        'ftk_EffectivePrice'
+        )
+    
+    # Combine ActualCost and AmortizedCost
+    $CombinedCost = $ActualCost + $AmortizedCost
+    Write-Host "Combined Cost Data Count: $($CombinedCost.Count)"
+    
+    # Perform transformations based on the version
+    switch ($Version) {
+        "0.5" {
+            if ($null -ne $CombinedCost -and $CombinedCost.Count -gt 0) {
+                $firstRow = $CombinedCost[0]
+                # If the first row of the data does not contain the billingAccountId or BillingAccountId column, throw an error
+                if (-not ($firstRow.PSObject.Properties.Name -contains 'billingAccountId' -or $firstRow.PSObject.Properties.Name -contains 'BillingAccountId')) {
+                    throw "Cost details schema not supported."
+                }
+            # else throw an error 
+            } else {
+                throw "Combined Cost data is empty or null."
+            }
+            # TODO: Filter the actual cost data to rows where ChargeType == "Purchase" and PricingModel == "Reservation" or "SavingsPlan" (leave amortized cost as-is).
+            # This to ensure that we only have rows with Purchase, reservation or savings plan data. Only applicable to actual cost data. Leaving the amortized cost data as-is.
+
+            # 1. Uppercase Column Names Modifications
+            $CombinedCost = $CombinedCost | ForEach-Object {
+                $obj = $_
+                $newObj = New-Object PSObject
+                foreach ($property in $obj.PSObject.Properties) {
+                    # If property name starts with a lowercase letter, modify the name
+                    $newPropertyName = if ($property.Name -cmatch "^[a-z]") {
+                        $property.Name -creplace '^[a-z]', { $_.Value.ToUpper() }
+                    } else {
+                        # If not, keep the original name
+                        $property.Name
+                    }
+                    # Add the property to the new object with the potentially modified name
+                    $newObj | Add-Member -MemberType NoteProperty -Name $newPropertyName -Value $property.Value
+                }
+                # Return the new object with modified property names
+                $newObj
+                # write-host "newObj: $($newObj | Select-Object -First 1 | Out-String)"
+            }
+            
+            # Write-Host "Sample data after Uppercase Column Name Modifications: $($CombinedCost | Select-Object -First 1 | Out-String)"
+            # Count the number of rows in the data
+            Write-Host "Combined Cost Data Count after Uppercase Column Name Modifications: $($CombinedCost.Count)"
+
+
+            # 2. Column Name Modifications (prepend 'ftk_')
+            $CombinedCost = $CombinedCost | ForEach-Object {
```

Comment:
If looping thru every row is required, we might want to make all the changes in one run thru each row.